pkg/build/generator/generator.go

```diff
@@ -86,6 +92,29 @@ type streamRef struct {
 	tag string
 }
 
+// FetchServiceAccountSecrets retrieves the Secrets used for pushing and pulling
+// images from private Docker registries.
+func (g *BuildGenerator) FetchServiceAccountSecrets(namespace string) ([]kapi.Secret, error) {
+	// TODO: Check for the service account specified in the build config when it
+	//       will be added
+	if len(g.DefaultServiceAccountName) == 0 {
+		g.DefaultServiceAccountName = bootstrappolicy.BuilderServiceAccountName
```

Comment:
@liggitt what do you mean? initialize it somewhere else? we will have to do that everywhere when we call this method (or generateBuildConfig)... since we are now using just the BuilderServiceAccountName, is it OK to leave this as todo item for next pr?